### PR TITLE
LRSD-4121 - [CP] Create test for Customer User Permissions

### PR DESCRIPTION
**Pull request by abfranca**

### CODE REVIEW

**Comment by abfranca** [comment]
ci:test:sf

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/fixtures/customerPagesTest.ts [hunk 1]
```diff
@@ -6,11 +6,31 @@
...
 	homePage: async ({page}, use) => {
 		await use(new HomePage(page));
 	},
+	projectAttachmentsPage: async ({page}, use) => {
+		await use(new ProjectAttachmentsPage(page));
+	},
+	projectOverviewPage: async ({page}, use) => {
+		await use(new ProjectOverviewPage(page));
+	},
+	projectPaaSPage: async ({page}, use) => {
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/fixtures/customerPagesTest.ts:30]: Can we rename this to projectLiferayPaaSPage so it has the full name?

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/LoginPage.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+export class LoginPage {
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/LoginPage.ts:8]: Do the Liferay Portal tests have any tests that do a login this way? Or is this login specific to our site?

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	readonly heading: Locator;
+	readonly paasHeading: Locator;
+	readonly page: Page;
+	readonly subscriptionCard: Locator;
+
+	constructor(page: Page) {
+		this.heading = page.getByRole('heading', {name: 'Subscriptions'});
+		this.paasHeading = page.getByRole('heading', {name: 'Liferay PaaS'});
+		this.page = page;
+		this.subscriptionCard = page
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts:20]: Not sure if this should be in the generic overviewPage. At minimum it should be called "LiferayPaasProductionSubscriptionCard".

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export class ProjectOverviewPage {
+	readonly heading: Locator;
+	readonly paasHeading: Locator;
+	readonly page: Page;
+	readonly subscriptionCard: Locator;
+
+	constructor(page: Page) {
+		this.heading = page.getByRole('heading', {name: 'Subscriptions'});
+		this.paasHeading = page.getByRole('heading', {name: 'Liferay PaaS'});
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts:18]: Does this work if the account has multiple subscriptions? I think these headings turn into tabs, and then dropdowns (if there's more than a certain amount)

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+		await expect(projectOverviewPage.paasHeading).toBeVisible();
+
+		await expect(projectOverviewPage.subscriptionCard).toBeVisible();
+
+		await projectPaaSPage.goto(accountExternalReferenceCode);
+
+		await expect(projectPaaSPage.heading).toBeVisible();
+
+		await expect(projectPaaSPage.projectName).toBeVisible();
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts:156]: we dont need to look for the project name here

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/projectPaaSPage.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	readonly projectName: Locator;
+	readonly projectNotActivatedTag: Locator;
+
+	constructor(page: Page) {
+		this.finishActivationButton = page.getByRole('button', {
+			name: 'Finish Activation Icon order-',
+		});
+		this.heading = page.getByRole('heading', {name: 'Activation Status'});
+		this.page = page;
+		this.projectName = page.getByRole('heading', {
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/projectPaaSPage.ts:23]: this specific project name isn't tied to this page so we dont need it here

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts (hunk 3 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts:20]: Not always will be cards in this page? That's it? But sure, a better name is needed.

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts (hunk 4 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts:18]: I don't know, but i can test this.

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+		await projectTeamMembersPage.goto(accountExternalReferenceCode);
+
+		await expect(page.getByText(userEmailAddress)).toBeVisible();
+
+		const userAccount =
+			await apiHelpers.headlessAdminUser.getUserAccountByEmailAddress(
+				userEmailAddress
+			);
+
+		await apiHelpers.headlessAdminUser.patchUserAccount(userAccount, {
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts:131]: is this changing the user password from test to test? if it's a new user, should currentPassword just be blank?

Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts:131]: It's a new user, probable will work just passing the password, but will test to make sure.

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/fixtures/customerPagesTest.ts (hunk 1 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/fixtures/customerPagesTest.ts]: Done!

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts (hunk 3 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts]: Oh, i see it now... I've removed this one.

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts (hunk 4 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/ProjectOverviewPage.ts]: You were wright, i've also removed this one.

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts (hunk 5 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts]: Done!

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/projectPaaSPage.ts (hunk 6 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/pages/projectPaaSPage.ts]: Done!

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts (hunk 7 above)
Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts:131]: Worked fine just with password attribute.

**Comment by abfranca** [comment]
ci:test:sf

File: modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+		await expect(page.getByText(userEmailAddress)).toBeVisible();
+
+		await expect(projectTeamMembersPage.inviteButton).not.toBeVisible();
+
+		await expect(
+			projectTeamMembersPage.userActionColumnHeader
+		).not.toBeVisible();
+
+		await customerPerformLogout(page);
```
Comment (amosfong) [inline modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts:179]: i see you're logging out and back in as admin. Is this so the cleanup.after logic works correctly?

Comment (abfranca) [reply modules/test/playwright/tests/workspaces/liferay-customer-workspace/tests/general/unit/projectUserPermission.spec.ts:179]: yes, if don't it doesn't work.

**Pull request closed without merging**